modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -206,6 +215,8 @@ public int partitionId() {
      * @param indexDescriptor Index descriptor.
      */
     public PageMemoryHashIndexStorage getOrCreateHashIndex(StorageHashIndexDescriptor indexDescriptor) {
+        assert !sortedIndexes.containsKey(indexDescriptor.id());
```

Comment:
Will there be races between the two maps?

Reply:
No, since in real world no indexes should have the same ID. In case of an error - yes, but this is a sanity assertion after all